python/tests/test_highlevel.py

```diff
@@ -541,6 +561,150 @@ def test_samples(self):
             ]
         assert total == ts.num_samples
 
+    @pytest.mark.parametrize("time", [0, 0.1, 1 / 3, 1 / 4, 5 / 7])
+    def test_samples_time(self, time):
+        ts = self.get_tree_sequence(num_demes=2, n=20, times=[time])
+        assert np.array_equal(get_samples(ts, time=time), ts.samples(time=time))
+        for population in (None, 0):
+            assert np.array_equal(
+                get_samples(ts, time=time, population=population),
+                ts.samples(time=time, population=population),
+            )
+
+    @pytest.mark.parametrize(
+        "time_interval",
+        [
+            [0, 0.1],
+            (0, 1 / 3),
+            np.array([1 / 4, 2 / 3]),
+            (0.345, 5 / 7),
+            (-1, 1),
+        ],
+    )
+    def test_samples_time_interval(self, time_interval):
+        rng = np.random.default_rng(seed=931)
+        times = rng.uniform(low=time_interval[0], high=time_interval[1], size=20)
```

Comment:
hm, actually your test here is always returning all the samples, no? shouldn't maybe the times be uniform on `[0, 2*time_interval[1]]` or something?

Reply:
oops!